src/macos/liquid_glass.rs

```diff
@@ -1,28 +1,93 @@
 use std::{ffi::c_void, ptr::NonNull};
 
-use objc2::MainThreadMarker;
+use objc2::{
+    ffi::{objc_getAssociatedObject, objc_setAssociatedObject},
+    rc::Retained, runtime::AnyObject, MainThreadMarker,
+};
 use objc2_app_kit::{
-    NSAppKitVersionNumber, NSAutoresizingMaskOptions, NSColor, NSGlassEffectViewStyle, NSView,
-    NSWindowOrderingMode,
+    NSAppKitVersionNumber, NSAutoresizingMaskOptions, NSBox, NSBoxType, NSColor,
+    NSGlassEffectView, NSGlassEffectViewStyle, NSView, NSWindowOrderingMode,
 };
-use objc2_foundation::NSInteger;
+use objc2_foundation::{NSInteger, NSRect};
 
 use crate::{macos::ns_glass_effect_view_tagged::NSGlassEffectViewTagged, Error};
 
 /// NSView::tag for NSVisualEffectViewTagged, just a random number
 pub const NS_VIEW_TAG_GLASS_VIEW: NSInteger = 96945937;
 
+const MOVED_CONTENT_KEY: &str = "WindowVibrancyMovedContentKey";
+const BACKGROUND_VIEW_KEY: &str = "WindowVibrancyBackgroundViewKey";
+
+const OBJC_ASSOCIATION_ASSIGN: usize = 0x0;
+const OBJC_ASSOCIATION_RETAIN: usize = 0x301;
+
+/// Minimum NSAppKitVersionNumber for liquid glass support (macOS 15.0+)
+const MIN_APPKIT_VERSION_LIQUID_GLASS: f64 = 2685.0;
+
+#[derive(Debug, Clone)]
+pub struct LiquidGlassOptions {
+    pub(crate) style: super::NSGlassEffectViewStyle,
+    pub(crate) tint_color: Option<crate::Color>,
+    pub(crate) radius: Option<f64>,
+    pub(crate) opaque: Option<bool>,
+    pub(crate) content_view: Option<NonNull<NSView>>,
+}
+
+impl LiquidGlassOptions {
+    pub fn new(style: super::NSGlassEffectViewStyle) -> Self {
+        Self {
+            style,
+            tint_color: None,
+            radius: None,
+            opaque: None,
+            content_view: None,
+        }
+    }
+
+    pub fn tint_color(mut self, color: crate::Color) -> Self {
+        self.tint_color = Some(color);
+        self
+    }
+
+    pub fn radius(mut self, radius: f64) -> Self {
+        self.radius = Some(radius);
+        self
+    }
+
+    pub fn opaque(mut self, opaque: bool) -> Self {
+        self.opaque = Some(opaque);
+        self
+    }
+
+    pub fn content_view(mut self, view: NonNull<c_void>) -> Self {
+        self.content_view = Some(view.cast::<NSView>());
+        self
+    }
+}
+
+impl Default for LiquidGlassOptions {
+    fn default() -> Self {
+        Self::new(super::NSGlassEffectViewStyle::Regular)
+    }
+}
+
 pub unsafe fn apply_liquid_glass(
     ns_view: NonNull<c_void>,
```

Comment:
i know this is my bad but i think we can take an NSView here as well - the NonNull thing is from before we had the objc2 crates